### PR TITLE
Added csh-bootstrap to frontend

### PR DESCRIPTION
**Pull request by gambogi**
The project is now included as a git subtree. (That's where all the @bencentra commits are coming from)
Additionally, the base template was extracted from the haskell source file it was embedded in.

### CODE REVIEW

File: src/CSH/Eval/Frontend.hs [hunk 1]
```diff
@@ -63,8 +54,7 @@ getEvalAPI _ = WaiSubsite evalAPI
 
 -- | An example Html handler for the frontend. 
 getHomeR :: Handler Html
-getHomeR = defaultLayout $(whamletFile "frontend/index.hamlet")
```
Comment (sdemos) [inline src/CSH/Eval/Frontend.hs:66]: `defaultLayout` is set above to `evalLayout`. Even thought the two definitions are the same, I would say we should keep it as `defaultLayout`.

**Comment by sdemos** [comment]
Otherwise the pull request looks good.

**Comment by TravisWhitaker** [comment]
The addition of files with the names "Gruntfile.hs" and "package.json" smells a lot like needing Node.JS in the build system. If that's a potential consequence of this PR then I think we should discuss whether or not we want to incur those additional build-time dependencies.

To be clear, by "included as a git subtree" you mean you used `git read-tree`, right? Are we going to stay in-sync with upstream csh-bootstrap, applying some evals-specific changes? Apologies if this was discussed and I simply missed out, but I'm not clear on what we're trying to accomplish by doing this.

**Comment by sdemos** [comment]
@bencentra can correct me if I'm wrong at any point in this explanation. The package.json is because the csh-bootstrap projects is also an npm package, and that is the packaging metadata. The Gruntfile.js is the grunt build system information for compiling less to css, which also needs some npm packages to perform. These dependencies would only be needed by someone changing the csh-bootstrap files. It won't pull node.js into our project, and the grunt build won't be run by our build, but instead be run as needed if an update to the less files in csh-bootstrap is done. 

In theory, we don't need any of the extra scaffolding at all, and can just do with the css file output by the grunt build system. I would generally prefer for us to not put any changes into the csh-bootstrap css/less files, but rather have our own set of css (or rather, lucius) files that contain any of our changes to the style. The csh-bootstrap styles are just for having a consistent style across CSH systems.

**Comment by bencentra** [comment]
@sdemos you're 99% right. It's not an actual npm package (I didn't publish it), package.json is just used to list all dependencies, in this case the various Grunt plugins being used. Otherwise, spot on.

You don't need to have any of the csh-bootstrap source files in your project. Updates will be few and far between, and (to maintain compatibility with Bootstrap) will be non-breaking anyway, so you can just manually update the file whenever an update does happen. (The one update I can think of is when Marc Billow finishes his "CSH Style Guide" and I make the csh-bootstrap colors match the ones he chose.)

If you're gonna use Bootstrap, use csh-bootstrap to keep it CSH-y. Any additional changes will be done in your project's lucius files. Basically: csh-eval CSS > csh-bootstrap CSS > base Bootstrap CSS.

**Pull request merged**